### DISCUSSION TITLE
Error with the basic example code

### DISCUSSION THREAD

**Discussion by tmokmss** [asg017/sqlite-vss#84] [Q&A] [answered]
Hi, I wrote the following Python code based on [the basic example doc](https://alexgarcia.xyz/sqlite-vss/getting-started.html#basic-example-2-dimensional-vectors), and getting an error.

```py
import sqlite3
import sqlite_vss

# https://alexgarcia.xyz/sqlite-vss/python.html
db = sqlite3.connect(":memory:")
db.enable_load_extension(True)
sqlite_vss.load(db)
db.enable_load_extension(False)

(version,) = db.execute("select vss_version()").fetchone()
print(version)
print(sqlite3.sqlite_version)

db.execute(
    """
create virtual table vss_demo using vss0(
  a(2)
);
"""
)

db.execute(
    """
insert into vss_demo(rowid, a)
  select
    value ->> 0 as rowid,
    value ->> 1 as a
  from json_each('
    [
      [ 1, [1.0, 3.0]   ],
      [ 2, [3.0, 1.0]   ],
      [ 3, [-2.0, -2.0] ],
      [ 4, [-4.0, 1.0]  ]
    ]
  ');
"""
)

db.execute(
    """
select
  rowid,
  distance
from vss_demo
where vss_search(a, json('[2.0, 2.0]'))
limit 1;
"""
)
```

The error I got:

```
$ poetry run python src/sqlite.py
v0.1.1
3.42.0
libc++abi: terminating with uncaught exception of type faiss::FaissException: Error in virtual void faiss::IndexFlat::search(idx_t, const float *, idx_t, float *, idx_t *, const SearchParameters *) const at /Users/alex/actions/sqlite-vss-runner/_work/sqlite-vss/sqlite-vss/vendor/faiss/faiss/IndexFlat.cpp:34: Error: 'k > 0' failed
[1]    88730 abort      poetry run python src/sqlite.py
```

Could you guide me what's wrong with this? It seems the `searchMax` is being zero here but idk why.

https://github.com/asg017/sqlite-vss/blob/66813745a9793a14b3f0d0bd28d0c430a52a0e89/src/sqlite-vss.cpp#L1096-L1101

**Comment by asg017** [comment, marked as answer]
Those docs are still a work-in-progress, but the problem here canbe solved with:

```python
db.commit()
```

after inserting into `vss_demo`. 


This is because all writes to `vss0` tables don't appear until a transaction is complete, which you sometimes have to explicitly do in Python.

Apologies for the confusion! Will be updating the docs (and will catch/handle that error since that isn't the best experience)


Thanks for the detailed report!

**Reply by tmokmss** [reply]
That worked, thank you so much!
